sdk/azidentity/workload_identity.go

```diff
@@ -83,6 +83,7 @@ func NewWorkloadIdentityCredential(options *WorkloadIdentityCredentialOptions) (
 		ClientOptions:              options.ClientOptions,
 		DisableInstanceDiscovery:   options.DisableInstanceDiscovery,
 	}
+	// TODO: this will incorrectly report ClientAssertionCredential in traces
```

Comment:
If we end up not needing the `clientName` param cited earlier, this problem goes away.

Reply:
Fixed.